src/pages/en/guides/content-collections.mdx

```diff
@@ -187,290 +173,250 @@ defineCollection({
+While not recommended, Astro allows you to [modify frontmatter directly](/en/guides/markdown-content/#modifying-frontmatter-programmatically) using remark or rehype plugins. You can access this modified frontmatter within Content Collections by using the `remarkPluginFrontmatter` property returned from `render()`:
 
 ```astro "{ remarkPluginFrontmatter }"
 ---
-import { getCollection } from 'astro:content';
-const blogPosts = await getCollection('blog');
+import { getEntryBySlug } from 'astro:content';
+const blogPost = await getEntryBySlug('blog', 'post-1');
+const { remarkPluginFrontmatter } = await blogPost.render();
 ---
-
-{blogPosts.map(async (post) => {
-  const { remarkPluginFrontmatter } = await post.render();
-  return <p>{post.data.title} — {remarkPluginFrontmatter.readingTime}</p>
-})}
+<!-- 
+  This example assumes `readingTime` was injected by a remark plugin.
+  See our "reading time" example for more details:
+  https://docs.astro.build/en/guides/markdown-content/#example-calculate-reading-time
+-->
+<p>{blogPost.data.title} — {remarkPluginFrontmatter.readingTime}</p>
 ```
 
-Assuming `readingTime` was injected ([see our reading time example](/en/guides/markdown-content/#example-calculate-reading-time)), it will be available on the `remarkPluginFrontmatter` object.
-
-:::caution
-Remark and rehype plugins will have access to the _raw_ Markdown or MDX document frontmatter. This means frontmatter will not reflect any changes or defaults applied by your `schema`.
-:::
-
 <details>
-<summary>**🙋 Why don't `getCollection()` and `getEntry()` contain these values?**</summary>
+<summary>**🙋 Why don't `getCollection()` and `getEntryBySlug()` contain these values?**</summary>
 
-The remark and rehype pipelines are only run when your content is **rendered.** This lets `render()` access anything generated by these plugins like injected frontmatter. To stay performant, `getCollection()` and `getEntry()` do not have this capability.
+The remark and rehype pipelines are only run when your content is **rendered.** This lets `render()` access anything generated by these plugins like injected frontmatter. To stay performant, `getCollection()` and `getEntryBySlug()` do not have this capability.
 
 </details>
 
-## Generating pages from content collections
 
-You can create pages based on your content collections using [dynamic routes](/en/core-concepts/routing/#dynamic-routes). 
+## Generating Routes from Content
 
-Use `getCollection()` inside a [`getStaticPaths()`](/en/reference/api-reference/#getstaticpaths) function to query your content entries and provide the `slug` parameter for each page. 
+Because content collections live outside of the `src/pages` directory, its up to you to create your [dynamic routes](/en/core-concepts/routing/#dynamic-routes) that will render your content and any collection entries to HTML.
```

Comment:
```suggestion
Because content collections live outside of the `src/pages/` directory, its up to you to create [dynamic routes](/en/core-concepts/routing/#dynamic-routes) that will render your content and any collection entries to HTML.
```